adsws/accounts/views.py

```diff
@@ -203,6 +204,111 @@ def post(self):
         return {'message': 'success'}, 200
 
 
+
+class UserInfoView(Resource):
+    """
+    Implements getting user info from session ID or token, it should be limited
+    to internal use only
+    """
+    decorators = [
+        ratelimit.shared_limit_and_check("500/43200 second", scope=scope_func),
+        oauth2.require_oauth('adsws:internal')
+    ]
+
+    def post(self):
+        """
+        This endpoint provides the full identifying data associated to a given
+        token (indicated via POST data 'access_token' parameter). Example:
+
+        curl -H 'authorization: Bearer:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
+            -H "Content-Type: application/json"
+            -d '{"acess_token":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"}'
+            -X POST
+            'https://dev.adsabs.harvard.edu/v1/accounts/session'
+        """
+        data = get_post_data(request)
+        token = None
+        session_id = None
+        if 'access_token' in data:
+            access_token = data['access_token']
+            token = OAuthToken.query.filter_by(access_token=access_token).first()
+            return self._translate(token, session_id)
+        else:
+            return {'message': 'Missing access_token parameter in POST data'}, 500
+
+    def get(self):
+        """
+        This endpoint provides the full identifying data associated to a given
+        session (sent via cookie). Example:
+
+        curl -H 'authorization: Bearer:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
+            -H 'cookie: session=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx...'
+            'https://dev.adsabs.harvard.edu/v1/accounts/session'
+
+        Notice that sessions are not server side, but client stored and server
+        signed to avoid user manipulation.
+        """
+        if 'session' in request.cookies:
```

Comment:
this suggests that the cookie is submitted with the request; it is not a parameter

i think it should be a parameter; you are passing either an oauth token or cookie or user_id; and you want to get back information about the user (you are not making the request *on behalf* of that user)